src/ServerTelemetryChannel/SamplingTelemetryProcessor.cs

```diff
@@ -3,32 +3,34 @@
     using System;
     using System.Collections.Generic;
     using System.Threading;
-    
+
     using Microsoft.ApplicationInsights.Channel;
     using Microsoft.ApplicationInsights.DataContracts;
     using Microsoft.ApplicationInsights.Extensibility;
     using Microsoft.ApplicationInsights.Extensibility.Implementation;
     using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel.Implementation;
-    
+
     /// <summary>
     /// Represents a telemetry processor for sampling telemetry at a fixed-rate before sending to Application Insights.
     /// </summary>
     public sealed class SamplingTelemetryProcessor : ITelemetryProcessor
     {
-        private const string DependencyTelemetryName = "Dependency";
-        private const string EventTelemetryName = "Event";
-        private const string ExceptionTelemetryName = "Exception";
-        private const string PageViewTelemetryName = "PageView";
-        private const string RequestTelemetryName = "Request";
-        private const string TraceTelemetryName = "Trace";
+        private const string DependencyTelemetryName = "DEPENDENCY";
+        private const string EventTelemetryName = "EVENT";
+        private const string ExceptionTelemetryName = "EXCEPTION";
+        private const string PageViewTelemetryName = "PAGEVIEW";
+        private const string RequestTelemetryName = "REQUEST";
+        private const string TraceTelemetryName = "TRACE";
 
         private readonly char[] listSeparators = { ';' };
         private readonly IDictionary<string, Type> allowedTypes;
 
-        private HashSet<Type> excludedTypesHashSet;
+        private readonly AtomicSampledItemsCounter proactivelySampledOutCounters = new AtomicSampledItemsCounter();
+
+        private SamplingTelemetryItemTypes excludedTypesFlags;
         private string excludedTypesString;
 
-        private HashSet<Type> includedTypesHashSet;
+        private SamplingTelemetryItemTypes includedTypesFlags;
```

Comment:
Why do we need both `included` and `excluded` flags?

Reply:
We use both in the default configuration - we set up one sampling with excluded events, another with only events included.

Reply:
In the default config, I thought those were two separate instances of SamplingTelemetryProcessor.

Is it ever valid for one Processor to have a set of both Included and Excluded?

Reply:
Correct, those are separate in the default config. So, it's more of a homage to the previous implementation rather than a requirement, I guess. They are currently both exposed as public properties on the configuration, so that part would stay the same. We can cook up internal implementation to only construct included list. Due to no perf gain at this (we remove one atomic check on `excluded.HasFlag`), not sure if I want this as part of this PR.